src/wfs/src/test/java/org/geoserver/wfs/response/ShapeZipTest.java

```diff
@@ -740,4 +752,159 @@ private String get4326_ESRI_WKTContent() {
                 + "PRIMEM[\"Greenwich\",0.0],"
                 + "UNIT[\"Degree\",0.0174532925199433]]";
     }
+
+    /** Test for Point ZM support on GetFeature shapefile output */
+    @Test
+    public void testPointZMShp() throws Exception {
+        // create a feature collection of POINT ZM (4D)
+        GeometryFactory gf = JTSFactoryFinder.getGeometryFactory();
+        SimpleFeatureType featureType =
+                DataUtilities.createType("pointmz", "name:String,geom:Point:4326");
+        SimpleFeatureBuilder fb = new SimpleFeatureBuilder(featureType);
+        fb.add("point1");
+        fb.add(gf.createPoint(new CoordinateXYZM(1, 2, 3, 4)));
+        List<SimpleFeature> features = new ArrayList<SimpleFeature>();
+        features.add(fb.buildFeature("1"));
+        SimpleFeatureCollection featureCollection = DataUtilities.collection(features);
+        // write the zip shapefile bytes
+        byte[] zipBytes = writeOut(featureCollection);
+        // get shp file bytes only
+        byte[] resultBytes = getShpOnlyBytes(zipBytes);
+        // get expected byte array
+        InputStream resource =
```

Comment:
The test works in its intention and does a valid test, but if it ever failed, how does one figure out where the difference is? They would have to open the file with some reader and figure out the differences (could there be "cosmetic" differences? I cannot remember, but file contents test seem to test more than just whether M is encoded or not). A different approach could be to read back the SHP, check the geometry type, read the geometry, check it has the expected contents.

Reply:
I agree, but reading of geometries from shape files with M measurements is not supported (no funded). I spend a bit of extra time extending ``PropertiesDataStore`` with some base code to read geometries with M coordinates and implement it for line strings. No spare time left for the shape files reading.

Reply:
Oh I see, no problem then.